libs/application/template-api-modules/src/lib/modules/template-api.service.ts

```diff
@@ -0,0 +1,66 @@
+import { Injectable } from '@nestjs/common'
+
+import { ApplicationTypes } from '@island.is/application/core'
+
+import { TemplateApiModuleActionProps } from '../types'
+
+import { ParentalLeaveService, ReferenceTemplateService } from './templates'
+
+interface ApplicationApiAction {
+  templateId: string
+  type: string
+  props: TemplateApiModuleActionProps
+}
+
+interface PerformActionEvent {
+  response: Error | string
+}
+
+type PerformActionResult = [boolean, PerformActionEvent?]
+
+@Injectable()
+export class TemplateAPIService {
+  constructor(
+    private readonly parentalLeaveService: ParentalLeaveService,
+    private readonly referenceTemplateService: ReferenceTemplateService,
```

Comment:
We should look into [ModuleRef](https://docs.nestjs.com/fundamentals/module-ref) and [custom provider tokens](https://docs.nestjs.com/fundamentals/custom-providers#non-class-based-provider-tokens) to simplify all of this. Not a blocker right now.

Reply:
This trick is also interesting: https://itnext.io/advanced-nestjs-techniques-part-1-custom-decorators-aa6d7f85c5b6?source=false---------0